Documentation/General/finalReport/overview.tex

```diff
@@ -1 +1,40 @@
-\chapter{Overview of the developed and implemented software product}
\ No newline at end of file
+\chapter{Overview of the developed and implemented software product}
+
+\section{Introduction}
+When entering MoodCat, a user can choose multiple of a list of moods to listen music to. Based on these moods a list of rooms
+sorted on relevance is shown, along with its title and which song is currently playing. When entering a room
+the user can chat with other users that have joined the room.
+
+The software of MoodCat consists of two main components: the Java based backend and the frontend which is a website,
+hosted at \url{http://moodcat.me}.
+
+\section{Frontend}
+The frontend uses Angular.js for managing its objects and connection with the backend and bootstrap
+for the layout and design.
+Angular.js is an MVC framework written in Javascript, next to providing a nice way to split
+and integrate with models, views and controllers, it also simplifies managing and updating displayed data.
+Angular reads HTML pages that contain custom tag attributes. The attributes are interpreted by angular, and
+replaced with actual data from the model.
+The model is in turn updated through constant interval based polling of the backend, which is made possible by the REST API from the backend
+and angular's capability of parsing it. When the time of the application is too much out of sync with the backend,
```

Comment:
This is actually a ("temporary") hack to simulate "streaming" of the music in rooms. Maybe describe that the music is "streamed" to the frontend, so that all users in the same room listen to the same music at the same time.